### DISCUSSION TITLE
How to get Ahead and Behind commit counts for a branch? Is there any API for the same.

### DISCUSSION THREAD

**Discussion by cp5364** [community/community#42292] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

I need to get the ahead and behind commit count of a branch, like in bitbucket. Is there any such API available?

**Comment by yaramostafa** [comment]
I guess you can find your answer here https://stackoverflow.com/questions/20433867/git-ahead-behind-info-between-master-and-branch

**Reply by cp5364** [reply]
Hi @yaramostafa . Thank you. But the link provides solution using git commands. But i am in search of any github API which gives me this data. I have checked in the official GitHub API documentation 'https://docs.github.com/en/rest?apiVersion=2022-11-28", but could not find what i was looking for.

**Comment by darwintree** [comment]
I found this. There is "ahead_by" and "behind_by" field in the response, and I think this is what you need
https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits

**Comment by nuri-yoo** [comment]
If you want a ready-made CLI that wraps this API, I built [branch-watch](https://github.com/nuri-yoo/branch-watch) for exactly this use case.

`bw branches owner/repo` shows every branch vs. the default branch with ahead/behind commit counts. `bw forks` does the same for all your forked repositories at once.

Uses `GET /repos/{owner}/{repo}/compare/{base}...{head}` under the hood.
